docs/rules/asi-safe.md

```diff
@@ -0,0 +1,33 @@
+# Avoid possible errors from automatic semicolon insertion (asi-safe)
+
+Semicolons are optional in JavaScript, see the documentation for [semi](./semi.md) for a fuller discussion of that feature.
+
+If you choose to take advantage of automatic semicolon insertion (ASI), then there are a few types of expression to avoid - those which are interpreted as continuing the previous line.
+
+## Rule Details
+
+This rule is aimed at ensuring consistent use of semicolons. You can decide whether or not to require semicolons at the end of statements.
+
+The following patterns are considered warnings:
+
+```js
+(1 || 2).someFunctionCall()
+
+[1, 2, 3].forEach(console.log)
+```
+
+The following patterns are not considered warnings:
+
+```js
+;(1 || 2).someFunctionCall()
+
+void [1, 2, 3].forEach(console.log)
+```
+
+## When Not To Use It
+
+If you are using the [semi](semi.md) rule, then this rule does not add any additional safety.
```

Comment:
I did mean that, but I now think that this rule should actually be on by default - as the cases it flags are not caught by semi: always

I wrote the docs before the latest implementation